src/libraries/System.Private.CoreLib/src/System/Text/Decoder.cs

```diff
@@ -261,6 +261,13 @@ public virtual unsafe int GetChars(ReadOnlySpan<byte> bytes, Span<char> chars, b
 
             bytesUsed = byteCount;
 
+            if (byteCount == 0)
```

Comment:
>             [](start = 0, length = 12)

There will be some concern I guess with this change which can cause infinite loops in the caller side if they didn't check before on the byte or char count and dependent on the exception especially when having flush = false.

Reply:
Is that a real concern and you think this should be closed?  Or just a hypothetical?

Reply:
With encoding I saw all weird things and I recall we ran into some infinite loop before in the full framework. So, I cannot be certain either way. if we want to go with this change, I would at least we document the change and use 6.0 release hopefully catch any bad usages of the API.